src/app/users/users-achievements/users-achievements.service.ts

```diff
@@ -1,19 +1,44 @@
 import { Injectable } from '@angular/core';
 import { CouchService } from '../../shared/couchdb.service';
+import { map } from 'rxjs/operators';
+import { ResourcesService } from '../../resources/resources.service';
+import { ActivatedRoute } from '@angular/router';
 
 @Injectable({
   providedIn: 'root'
 })
 export class UsersAchievementsService {
 
   readonly dbName = 'achievements';
+  resourceIds = [];
+  parent = this.route.snapshot.data.parent;
 
   constructor(
-    private couchService: CouchService
-  ) {}
+    private couchService: CouchService,
+    private resourcesService: ResourcesService,
+    private route: ActivatedRoute
+  ) {
+    this.resourcesService.resourcesListener(this.parent).pipe(map((resources: any) => {
+      resources.map((res: any) => { this.resourceIds.push(res._id); });
+      return this.resourceIds;
+    })).subscribe((resIds: any) => {
+      this.resourceIds = resIds;
+    });
+    this.resourcesService.requestResourcesUpdate(this.parent);
```

Comment:
There's a problem here: this will run once when the application starts.  If you delete a resource and immediately check your achievements, it will still be there.  If you open a different page that calls this, it will run and remove the resource.

I think you need to incorporate this into `getAchievements` so the request is made every time and the resources used is up to date.